modules/security/keyvault/main.bicep

```diff
@@ -0,0 +1,100 @@
+@description('Deployment Location')
+param location string
```

Comment:
I see a mix of some modules defaulting this value to the `resourceGroup().location` and some that don't. It would be nice if this was more consistent or set as a standard in the contributing guide.

Reply:
This one is tricky, and a weird edge case. It looks good to have the default as 'resourceGroup().location', but this resulted in issues when I tried to consume our modules in my Azure Marketplace Applications.

There are 3 linter rules that we have to comply with for the Marketplace.
https://learn.microsoft.com/en-us/azure/azure-resource-manager/bicep/linter-rule-no-hardcoded-location
https://learn.microsoft.com/en-us/azure/azure-resource-manager/bicep/linter-rule-no-loc-expr-outside-params
https://learn.microsoft.com/en-us/azure/azure-resource-manager/bicep/linter-rule-explicit-values-for-loc-params

Let me chat around and get some more input, and see if everyone is aligned with `param location string` or `param location string = resourceGroup().location`, in which case we need to get the Marketplace team to adjust their arm-ttk checks.